src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +136,24 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+    //This method can run into problems if the proper response isnt provided by the server info request.
+    public ServerInfo getInfo() {
+      try (final Connection connection = this.source.connect()) {
```

Comment:
I'm confused on what is going on in this method, specifically why the `'try'` block starts with `(final Connection connection = this.source.connect())`